EjercicioDiaTres.cs

```diff
@@ -0,0 +1,168 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace EjerciciosProyect_Visual.Net
+{
+    class EjercicioDiaTres
+    {
+        static EjercicioDiaTres ejercicioDia;
+        public static void Main(string[] args)
+        {
+            ejercicioDia = new EjercicioDiaTres();
+            ejercicioDia.llamarEjercicioUno();
+            ejercicioDia.llamarEjercicioDos();
+            ejercicioDia.llamarEjercicioTres();
+        }
+        public void llamarEjercicioUno()
+        {
+            int resultado;
+            Console.WriteLine("Ingresa el Primer número: ");
+            int numero1 = int.Parse(Console.ReadLine());
+            Console.WriteLine("Ingresa el Segundo número: ");
+            int numero2 = int.Parse(Console.ReadLine());
+            Console.WriteLine("Ingresa el Tercer número: ");
+            int numero3 = int.Parse(Console.ReadLine());
+            resultado = ejercicioDia.CalcularNumeroMayor(numero1, numero2, numero3);
+            Console.WriteLine("El Número Mayor de ellos es: " + resultado);
+            Console.ReadKey();
+        }
+        public void llamarEjercicioDos()
+        {
+            string nombredelmes;
+            Console.WriteLine("Digita un número del 1 al 12: ");
+            int numero = Convert.ToInt32(Console.ReadLine());
+            nombredelmes = ejercicioDia.DeterminarElMes(numero);
+            Console.WriteLine("El Número Digitado " + numero + " corresponde al Mes de: " + nombredelmes);
+            Console.ReadKey();
+        }
+        public void llamarEjercicioTres()
+        {
+            string numerosimpares;
+            Console.WriteLine("Ingrese un Número Positivo: ");
+            int numero = Convert.ToInt32(Console.ReadLine());
+            numerosimpares = ejercicioDia.MostrarNumerosImpares(numero);
+            Console.WriteLine("El Orden descendente al número " + numero + " es: " + numerosimpares);
+            Console.ReadKey();
+        }
+        public int CalcularNumeroMayor(int numero1, int numero2, int numero3)
+        {
+            int resultado;
+            if (numero1 >= numero2 && numero1 > numero3)
+            {
+                resultado = numero1;
+            }
+            else if (numero2 >= numero1 && numero2 > numero3)
+            {
+                resultado = numero2;
+            }
+            else
+            {
+                resultado = numero3;
+            }
+            return resultado;
+        }
+        public string DeterminarElMes(int numero)
```

Comment:
Esta logica esta bien, cumple con lo que se pide y se logra el resultado, sin embargo vamos a hablar de un tema que se llama *complejidad ciclomatica*, 

Qué es la complejidad ciclomatica de un algoritmo? 
mide el número de flujos distintos de ejecución que puede tener el código de un artefacto de software, dicho llanamente, nos dice cuantos ifs-then-else, while, for, switch…etc, tenemos en nuestro código.
Te dejaré algunos link de estudio de este tema para que lo leas y lo tengas presente:
- https://medium.com/@mpijierro/complejidad-ciclom%C3%A1tica-y-como-reducirla-7374c215f666
- http://joaquinoriente.com/2012/11/15/complejidad-ciclomatica-nuestro-codigo-es-facil-de-mantener-y-probar/


## Que vamos a hacer?

Cómo puedes observar tu tienes contando los if, else, un total de 24. lo que se convierte en un algoritmo con complejidad ciclomatica media-alta. asi que vamos a hacer un **refactor del codigo**, 

Qué es una refactorizacion de un codigo?
El proceso de cambiar un sistema de software de tal manera que no altera el comportamiento exterior del código pero mejora su estructura interna. Es una forma disciplinada de limpiar código que minimiza las posibilidades de introducir un error. En esencia, cuando refactorizas, estás mejorando el diseño del código después de que fue escrito.
- https://medium.com/@XFelipeM/refactorizaci%C3%B3n-de-c%C3%B3digo-like-a-champion-1-a9c9c95d704f

## Cómo lo vamos a hacer?
Vamos a rediseñar este algoritmo de modo que cumple con su resultado dado una entrada y para ello te voy a dejar este link 

https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/switch

El switch permite dado una entrada tomar una decision directamente sin necesidad de hacer condicionales

Algunos ejemplos de como implementarlo:
https://webprogramacion.com/135/csharp/instruccion-switch.aspx

Si necesitass mas informacion del switch consulta en google por ello.

Asi que vamos a refactorizar esta logica con el switch 😄

Reply:
Ay bendito... refactorizemos entonces...

Reply:
Refactorizado